### PR TITLE
Set security from key-management or guess if not available

### PR DESCRIPTION
**Pull request by cfconrad**
## Problem

Fixes: https://github.com/jcronenberg/agama/issues/81

### CODE REVIEW

File: rust/migrate-wicked/src/wireless.rs [hunk 1]
```diff
@@ -416,27 +425,93 @@ where
...
+        Ok(model::SecurityProtocol::WEP)
+    } else if network.wpa_eap.is_some() {
+        Ok(model::SecurityProtocol::WPA2Enterprise)
+    } else if network.wpa_psk.is_some() {
+        Ok(model::SecurityProtocol::WPA2)
+    } else {
+        Ok(model::SecurityProtocol::WEP)
+    };
+    log::warn!(
+        "Unsupported key-management protocol(s) '{}' guess '{}'",
```
Comment (jcronenberg) [inline rust/migrate-wicked/src/wireless.rs:469]:
```suggestion
        "Unsupported key-management protocol(s) '{}' guessing '{}'",
```

File: rust/migrate-wicked/src/wireless.rs [hunk 2]
```diff
@@ -639,4 +716,147 @@ mod tests {
...
+        net.key_management = vec!["wpa-eap".to_string(), "wpa-psk".to_string()];
+        assert!(!wireless_security_protocol(&net).is_ok());
+        net.key_management = vec!["wpa-eap".to_string(), "sae".to_string()];
+        assert!(!wireless_security_protocol(&net).is_ok());
+        net.key_management = vec!["wpa-eap-suite-b".to_string(), "sae".to_string()];
+        assert!(!wireless_security_protocol(&net).is_ok());
+        net.key_management = vec!["wpa-eap-suite-b".to_string(), "owe".to_string()];
+        assert!(!wireless_security_protocol(&net).is_ok());
+        net.key_management = vec!["wpa-psk".to_string(), "owe".to_string()];
+        assert!(!wireless_security_protocol(&net).is_ok());
```
Comment (jcronenberg) [inline rust/migrate-wicked/src/wireless.rs:809]: `!...().is_ok()` -> `...().is_err()`

File: rust/migrate-wicked/src/wireless.rs [hunk 3]
```diff
@@ -416,27 +425,93 @@ where
...
+    };
+    log::warn!(
+        "Unsupported key-management protocol(s) '{}' guessing '{}'",
+        mgmt.join(","),
+        match v {
+            Ok(s) => s.to_string(),
+            _ => "-".to_string(),
+        }
+    );
+    v
```
Comment (jcronenberg) [inline rust/migrate-wicked/src/wireless.rs:476]: I think it's nicer to not wrap `v` (maybe also use a more descriptive name) in a `Result` and instead do this
```suggestion
    log::warn!(
        "Unsupported key-management protocol(s) '{}' guessing '{}'",
        mgmt.join(","),
        v
    );
    Ok(v)
```

Comment (cfconrad) [reply rust/migrate-wicked/src/wireless.rs:476]: okay

**Review by jcronenberg — APPROVED** [review]

**Pull request merged**